velocity_modelling/scripts/plot_velocity_model.py

```diff
@@ -0,0 +1,224 @@
+        ),
+    ] = None,
+    resolution: Annotated[
+        Optional[float], typer.Option(help="Resolution of the velocity model")
+    ] = None,
+    centre_lat: Annotated[
+        Optional[float], typer.Option(help="Centre latitude", min=-90, max=90)
+    ] = None,
+    centre_lon: Annotated[
+        Optional[float], typer.Option(help="Centre longitude", min=-180, max=180)
+    ] = None,
+    bearing: Annotated[
+        Optional[float], typer.Option(help="Model bearing", min=0, max=360)
+    ] = None,
+    extent_x: Annotated[
+        Optional[float], typer.Option(help="Extent in x-direction", min=0)
+    ] = None,
+    extent_y: Annotated[
+        Optional[float], typer.Option(help="Extent in y-direction", min=0)
+    ] = None,
+    title: Annotated[Optional[str], typer.Option(help="Figure title")] = None,
+    latitude_pad: Annotated[
+        float, typer.Option(help="Latitude padding for figure (in degrees)", min=0)
+    ] = 0,
+    longitude_pad: Annotated[
+        float, typer.Option(help="Longitude padding for figure (in degrees)", min=0)
+    ] = 0,
+    dpi: Annotated[
+        float, typer.Option(help="Plot output DPI (higher is better)")
+    ] = 300,
+    component: Annotated[
+        VelocityModelComponent,
+        typer.Option(help="Velocity model component to overlay."),
+    ] = VelocityModelComponent.p_wave,
+    slice: Annotated[
+        int, typer.Option(help="z-level slice of velocity model to plot.")
+    ] = 0,
+    nz: Annotated[
+        Optional[int], typer.Option(help="Number of z-slices in velocity model")
+    ] = None,
+    transparency: Annotated[
+        int, typer.Option(help="Velocity model overlay transparency, (0 = opaque)")
+    ] = 80,
+):
+    """Plot the boundary of the velocity model."""
+    nx = ny = None
+    if vm_params_ffp:
+        with open(vm_params_ffp) as vm_params_handle:
+            vm_params = yaml.safe_load(vm_params_handle)
+
+        # Extent x and extent y are swapped in meaning between the old
+        # vm params and the new bounding box class. So we swap them
+        # here when we load them.
+        box = bounding_box.BoundingBox.from_centroid_bearing_extents(
+            [vm_params["MODEL_LAT"], vm_params["MODEL_LON"]],
+            vm_params["MODEL_ROT"],
+            vm_params["extent_y"],
+            vm_params["extent_x"],
+        )
+
+        resolution = vm_params.get("hh")
+        nx = vm_params.get("nx")
+        ny = vm_params.get("ny")
+        nz = vm_params.get("nz")
+
+    elif all(
+        parameter is not None
+        for parameter in [
+            centre_lat,
+            centre_lon,
+            bearing,
+            extent_x,
+            extent_y,
+        ]
+    ):
+        box = bounding_box.BoundingBox.from_centroid_bearing_extents(
+            [centre_lat, centre_lon], bearing, extent_x, extent_y
+        )
+        if resolution:
+            nx = int(round(extent_x / resolution))
+            ny = int(round(extent_y / resolution))
```

Comment:
@sungeunbae relating to my comment on the value of the `Optional` type. If I leave out the `if resolution:` statement and instead have code like 

```python
       box = bounding_box.BoundingBox.from_centroid_bearing_extents(
            [centre_lat, centre_lon], bearing, extent_x, extent_y
        )
        nx = int(round(extent_x / resolution))
        ny = int(round(extent_y / resolution))
```

then the mypy type-checker integrated into my editor config gives me the error

```
plot_velocity_model.py 117 error   mypy [operator]: Unsupported right operand type for / ("None")
```

Correctly informing me that `resolution = None` is possible and will crash if it happened. So I get informed of the bug in my code early and don't find it later when someone runs my script and complains.